hnn_core/feed.py

```diff
@@ -36,251 +70,207 @@ class ExtFeed(object):
+        target_syn_weights_zero = (params[target_cell_type][0] <= 0.0 and
+                                   params[target_cell_type][1] <= 0.0)
+
+    all_syn_weights_zero = True
+    for key in params.keys():
+        if key.startswith(('L2Pyr', 'L5Pyr', 'L2Bask', 'L5Bask')):
+            if params[key][0] > 0.0:
+                all_syn_weights_zero = False
+
+    event_times = list()
+    if feed_type == 'extpois' and not target_syn_weights_zero:
+        event_times = _create_extpois(
+            t0=params['t_interval'][0],
+            T=params['t_interval'][1],
+            # ind 3 is frequency (lamtha))
+            lamtha=params[target_cell_type][3],
+            prng=prng)
+    elif feed_type.startswith(('evprox', 'evdist')) and \
+            target_cell_type in params:
+        event_times = _create_gauss(
+            mu=params['t0'],
+            # ind 3 is sigma_t (stdev))
+            sigma=params[target_cell_type][3],
+            numspikes=int(params['numspikes']),
+            prng=prng)
+    elif feed_type == 'extgauss' and not target_syn_weights_zero:
+        event_times = _create_gauss(
+            mu=params[target_cell_type][3],
+            sigma=params[target_cell_type][4],
+            numspikes=50,
+            prng=prng)
+    elif feed_type == 'common' and not all_syn_weights_zero:
+        event_times = _create_common_input(
+            distribution=params['distribution'],
+            t0=params['t0'],
+            t0_stdev=params['t0_stdev'],
+            tstop=params['tstop'],
+            f_input=params['f_input'],
+            stdev=params['stdev'],
+            repeats=params['repeats'],
+            events_per_cycle=params['events_per_cycle'],
+            prng=prng,
+            prng2=prng2)
+
+    # brute force remove non-zero times. Might result in fewer vals
+    # than desired
+    # values MUST be sorted for VecStim()!
+    if len(event_times) > 0:
+        event_times = event_times[event_times > 0]
+        event_times.sort()
+        event_times = event_times.tolist()
+
+    return event_times
+
+
+def _create_extpois(t0, T, lamtha, prng):
+    """Create poisson inputs.
+
+    Parameters
+    ----------
+    t0 : float
+        The start time (in ms).
+    T : float
+        The end time (in ms).
+    lamtha : float
+        The rate parameter for spike train (in Hz)
+    prng : instance of RandomState
+        The random state.
+
+    Returns
+    -------
+    event_times : array
+        The event times.
+    """
+    # see: http://www.cns.nyu.edu/~david/handouts/poisson.pdf
+    if t0 < 0:
+        raise ValueError('The start time for Poisson inputs must be'
+                         f'greater than 0. Got {t0}')
+    if T < t0:
+        raise ValueError('The end time for Poisson inputs must be'
+                         f'greater than start time. Got ({t0}, {T})')
+    if lamtha <= 0.:
+        raise ValueError(f'Rate must be > 0. Got {lamtha}')
+
+    event_times = list()
+    t_gen = t0
```

Comment:
doing:

```py
t_gen = t0 + prng.exponential(1. / lamtha) * 1000.
```

will revert to old behavior @rythorpe

Reply:
Do need to address anything here @jasmainak or was this addressed above?

Reply:
I think we're good here